crates/l3/appchain-core-contract-client/src/interfaces/operator.rs

```diff
@@ -0,0 +1,103 @@
+use color_eyre::{eyre::eyre, Result};
+use appchain_utils::LocalWalletSignerMiddleware;
+use appchain_utils::{call_contract, invoke_contract};
+use starknet_accounts::ConnectedAccount;
+use starknet_core::types::{FieldElement, InvokeTransactionResult};
+use starknet_providers::{
+    jsonrpc::{HttpTransport, JsonRpcClient},
+    Provider,
+};
+
+pub struct Operator<'a> {
+    signer: &'a LocalWalletSignerMiddleware,
+    address: FieldElement,
+}
+
+impl<'a> Operator<'a> {
+    pub fn new(address: FieldElement, signer: &'a LocalWalletSignerMiddleware) -> Self {
+        Self { signer, address }
+    }
+
+    fn provider(&self) -> &JsonRpcClient<HttpTransport> {
+        self.signer.provider()
+    }
+
+    pub async fn register_operator(
+        &self,
+        new_operator: FieldElement,
+    ) -> Result<InvokeTransactionResult> {
+        invoke_contract(
+            &self.signer,
```

Comment:
`self.signer` is already a ref. I'm not sure you need to reference it again given the signature expect a `signer: &LocalWalletSignerMiddleware`

Reply:
Check it and if so, fix it for all others occurrences too

Reply:
fixed